pool/app/filter/repo/repo.ml

```diff
@@ -226,13 +234,102 @@ module Sql = struct
       | Bool _ | Date _ | Language _ | Nr _ | Option _ -> None)
   ;;
 
+  let create_temporary_invitation_table query =
+    let open Dynparam in
+    let open Caqti_request.Infix in
+    let create_request ids =
+      Format.asprintf
+        {sql|
+        CREATE TEMPORARY TABLE tmp_invitations
+        (INDEX contact_index (contact_uuid),
+         INDEX experiment_index (experiment_uuid)
+        )
+        AS (
+          SELECT
+            pool_invitations.contact_uuid AS contact_uuid,
+            pool_invitations.experiment_uuid AS experiment_uuid
+          FROM pool_invitations
+          WHERE experiment_uuid IN ( %s ))
+        |sql}
+        (CCList.mapi
+           (fun i _ -> Format.asprintf "UNHEX(REPLACE($%n, '-', ''))" (i + 1))
+           ids
+         |> CCString.concat ",")
+    in
+    let open CCOption in
+    let fnc connection =
+      query
+      >|= Repo_utils.find_experiments_by_key Key.Invitation
+      |> function
+      | None | Some [] -> Lwt_result.return ()
+      | Some ids ->
+        let (Pack (pt, pv)) =
+          CCList.fold_left
+            (fun dyn id -> dyn |> add Caqti_type.string id)
+            empty
+            ids
+        in
+        let (module Connection : Caqti_lwt.CONNECTION) = connection in
+        let request = create_request ids |> pt ->. Caqti_type.unit in
+        Connection.exec request pv
+    in
+    fnc
+  ;;
+
+  let create_temporary_assignments_table query =
+    let open Dynparam in
+    let open Caqti_request.Infix in
+    let create_request ids =
+      Format.asprintf
+        {sql|
+        CREATE TEMPORARY TABLE tmp_assignments 
+        (INDEX contact_index (contact_uuid),
+         INDEX experiment_index (experiment_uuid)
+        )
+        AS (
+          SELECT
+            pool_assignments.contact_uuid AS contact_uuid,
+            pool_sessions.experiment_uuid AS experiment_uuid
+          FROM pool_assignments
+            INNER JOIN pool_sessions ON pool_sessions.uuid = pool_assignments.session_uuid
+          WHERE experiment_uuid IN ( %s ))
```

Comment:
@mabiede this is the key of the feature really, creating this table just by the fact that there are assignments with sessions for this experiment/contact combo.